Py_FS/wrapper/nature_inspired/ALO.py

```diff
@@ -0,0 +1,166 @@
+            for j in range (0, population.shape[1] - 1):
+                minimum_c_i[0,j]   = antlions[0,j]/i_ratio
+                maximum_d_i[0,j]   = antlions[-1,j]/i_ratio
+                elite_antlion[0,j] = antlions[0,j]
+                minimum_c_e[0,j]   = antlions[0,j]/i_ratio
+                maximum_d_e[0,j]   = antlions[-1,j]/i_ratio
+                rand = np.random.random()
+                if (rand < 0.5):
+                    minimum_c_i[0,j] =   minimum_c_i[0,j] + antlions[ant_lion,j]
+                    minimum_c_e[0,j] =   minimum_c_e[0,j] + elite_antlion[0,j]
+                else:
+                    minimum_c_i[0,j] = - minimum_c_i[0,j] + antlions[ant_lion,j]
+                    minimum_c_e[0,j] = - minimum_c_e[0,j] + elite_antlion[0,j]
+
+                rand = np.random.random()
+                if (rand >= 0.5):
+                    maximum_d_i[0,j] =   maximum_d_i[0,j] + antlions[ant_lion,j]
+                    maximum_d_e[0,j] =   maximum_d_e[0,j] + elite_antlion[0,j]
+                else:
+                    maximum_d_i[0,j] = - maximum_d_i[0,j] + antlions[ant_lion,j]
+                    maximum_d_e[0,j] = - maximum_d_e[0,j] + elite_antlion[0,j]
+                x_random_walk = self.random_walk(iterations)
+                e_random_walk = self.random_walk(iterations)
+                min_x, max_x = min(x_random_walk), max(x_random_walk)
+                x_random_walk[count] = (((x_random_walk[count] - min_x)*(maximum_d_i[0,j] - minimum_c_i[0,j]))/(max_x - min_x)) + minimum_c_i[0,j]
+                min_e, max_e = min(e_random_walk), max(e_random_walk)
+                e_random_walk[count] = (((e_random_walk[count] - min_e)*(maximum_d_e[0,j] - minimum_c_e[0,j]))/(max_e - min_e)) + minimum_c_e[0,j]
+                population[i,j]=(x_random_walk[count] + e_random_walk[count])/2
+        return population, antlions
+    
+    
+    
+    
+    
+    def next(self):
+        print('\n================================================================================')
+        print('\n                          Iteration - {}'.format(self.cur_iter + 1))
+        print('================================================================================\n')
+            
+        self.ants, self.population = self.update_ants(self.ants, self.population,self.cur_iter,self.max_iter,self.weight_acc)
+        self.ants, self.population = self.combine(self.ants, self.population)
+        for i in range(self.num_agents):
+            for j in range(self.train_data.shape[-1]):
+                trans_value=self.trans_function(self.population[i,j])
+                if (np.random.random() < trans_value): 
+                    self.population[i,j] = 1
+                else:
+                    self.population[i,j] = 0
+        self.cur_iter+=1
+        compute_fitness(self.weight_acc)
+        
+        
+        
+
+if __name__ == '__main__':
+    data=datasets.load_digits()
+    alo = ALO(num_agents=50,
```

Comment:
Change the number of agents to 20. This is because all other algos have 20.

Reply:
Done!